hw/xbox/smbus_xbox_smc.c

```diff
@@ -178,9 +179,43 @@ static uint8_t smc_receive_byte(SMBusDevice *dev)
     return 0;
 }
 
+bool smbus_avpack_to_reg(char *avpack, uint8_t *value)
+{
+    if (strcmp(avpack, "composite") == 0) {
+        if (value) *value = SMC_REG_AVPACK_COMPOSITE;
+        return true;
+    }
+    if (strcmp(avpack, "scart") == 0) {
+        if (value) *value = SMC_REG_AVPACK_SCART;
+        return true;
+    }
+    if (strcmp(avpack, "svideo") == 0) {
+        if (value) *value = SMC_REG_AVPACK_SVIDEO;
+        return true;
+    }
+    if (strcmp(avpack, "vga") == 0) {
+        if (value) *value = SMC_REG_AVPACK_VGA;
+        return true;
+    }
+    if (strcmp(avpack, "rfu") == 0) {
+        if (value) *value = SMC_REG_AVPACK_RFU;
+        return true;
+    }
+    if (strcmp(avpack, "hdtv") == 0) {
+        if (value) *value = SMC_REG_AVPACK_HDTV;
+        return true;
+    }
+    if (strcmp(avpack, "none") == 0) {
+        if (value) *value = SMC_REG_AVPACK_NONE;
+        return true;
+    }
+    return false;
+}
```

Comment:
nit: Possible alternative for your consideration (I personally consider this cleaner):

```c
bool smbus_avpack_to_reg(char *avpack, uint8_t *value)
{
    uint8_t r;
    if (strcmp(avpack, "composite") == 0) {
        r = SMC_REG_AVPACK_COMPOSITE;
    } else if (strcmp(avpack, "scart") == 0) {
        r = SMC_REG_AVPACK_SCART;
    } else if (strcmp(avpack, "svideo") == 0) {
        r = SMC_REG_AVPACK_SVIDEO;
    } else if (strcmp(avpack, "vga") == 0) {
        r = SMC_REG_AVPACK_VGA;
    } else if (strcmp(avpack, "rfu") == 0) {
        r = SMC_REG_AVPACK_RFU;
    } else if (strcmp(avpack, "hdtv") == 0) {
        r = SMC_REG_AVPACK_HDTV;
    } else if (strcmp(avpack, "none") == 0) {
        r = SMC_REG_AVPACK_NONE;
    } else {
        return false;
    }

    if (value) *value = r;
    return true;
}
```

*(I personally also prefer `!strcmp`; I'm not sure what we do in the rest of the codebase)*

Reply:
Agree except `if (value) *value = r;` should have braces per qemu coding conventions

Reply:
^ it's actually GNU code style, and also it's a huge nitpick :stuck_out_tongue: 

I'd leave it as is, because style like this was also used on line 109. :grin:

Reply:
Two wrongs and all :)